includes/init_includes/init_customer_auth.php

```diff
@@ -79,27 +79,34 @@
  * 2 = customer authorization pending can browse but no prices
  */
 switch (true) {
+  /**
+   * bypass redirects for these scripts, to processing regardless of store mode or cust auth mode
+   */
+  case (preg_match('/ipn_main_handler\.php$/', $_SERVER['SCRIPT_NAME'])):

```

Comment:
When it all comes down to it, I'm good with this change either way. Was just curious if restricting the checks to the files shipped with Zen Cart might help avoid mistakes in 3rd party code...

But as you point out, some 3rd party code need to allow access to a PHP script when customer approval is not yet known (or pending). I'm all for adding some sort of hook in the future.

:+1: